src/sort-filter-bar/sort-filter-bar.ts

```diff
@@ -0,0 +1,747 @@
+
+    if (changed.has('dateSortSelectorVisible')) {
+      this.setupEscapeListeners();
+    }
+
+    if (changed.has('resizeObserver')) {
+      const oldObserver = changed.get(
+        'resizeObserver'
+      ) as SharedResizeObserverInterface;
+      if (oldObserver) this.disconnectResizeObserver(oldObserver);
+      this.setupResizeObserver();
+    }
+  }
+
+  private setupEscapeListeners() {
+    if (this.dateSortSelectorVisible) {
+      document.addEventListener(
+        'keydown',
+        this.boundDateSelectorEscapeListener
+      );
+    } else {
+      document.removeEventListener(
+        'keydown',
+        this.boundDateSelectorEscapeListener
+      );
+    }
+  }
+
+  private boundDateSelectorEscapeListener = (e: KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      this.dateSortSelectorVisible = false;
+    }
+  };
+
+  disconnectedCallback(): void {
+    if (this.resizeObserver) {
+      this.disconnectResizeObserver(this.resizeObserver);
+    }
+  }
+
+  private disconnectResizeObserver(
+    resizeObserver: SharedResizeObserverInterface
+  ) {
+    resizeObserver.removeObserver({
+      target: this.sortSelectorContainer,
+      handler: this,
+    });
+
+    resizeObserver.removeObserver({
+      target: this.desktopSortSelector,
+      handler: this,
+    });
+  }
+
+  private setupResizeObserver() {
+    if (!this.resizeObserver) return;
+    this.resizeObserver.addObserver({
+      target: this.sortSelectorContainer,
+      handler: this,
+    });
+
+    this.resizeObserver.addObserver({
+      target: this.desktopSortSelector,
+      handler: this,
+    });
+  }
+
+  private get mobileSelectorVisible() {
+    return this.selectorBarContainerWidth - 10 < this.desktopSelectorBarWidth;
+  }
+
+  private get alphaBarTemplate(): TemplateResult | typeof nothing {
+    if (!['title', 'creator'].includes(this.selectedSort)) return nothing;
+
+    if (this.alphaSelectorVisible === null) {
+      if (this.selectedSort === 'creator') return this.creatorSelectorBar;
+      if (this.selectedSort === 'title') return this.titleSelectorBar;
+    } else {
+      return this.alphaSelectorVisible === 'creator'
+        ? this.creatorSelectorBar
+        : this.titleSelectorBar;
+    }
+
+    return nothing;
+  }
+
+  handleResize(entry: ResizeObserverEntry): void {
+    if (entry.target === this.desktopSortSelector) {
+      this.desktopSelectorBarWidth = entry.contentRect.width;
+    } else if (entry.target === this.sortSelectorContainer) {
+      this.selectorBarContainerWidth = entry.contentRect.width;
+    }
+  }
+
+  private get sortDirectionSelectorTemplate() {
+    return html`
+      <div id="sort-direction-selector">
+        <button
+          id="sort-ascending-btn"
+          class="sort-button ${this.sortDirection === 'asc' ? 'selected' : ''}"
+          ?disabled=${this.selectedSort === 'relevance'}
+          @click=${() => {
+            this.setSortDirections('asc');
+          }}
+        >
+          ${sortIcon}
```

Comment:
yah, we can use image for this.

Possible Blocker: we may need an "sr-only" label for some of the icon filled buttons